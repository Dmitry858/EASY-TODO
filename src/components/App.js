import React from 'react';
import { connect } from 'react-redux';
import { Router, Route, Switch } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import qhistory from 'qhistory';
import { stringify, parse } from 'qs';
import Header from './Header/Header';
import Footer from './Footer';
import Home from './Home';
import About from './About';
import Lists from './Lists';
import List from './List';
import Archive from './Archive';
import Analytics from './Analytics';
import Help from './Help';
import Settings from './Settings';
import NotFound from './NotFound';
import getCookie from '../utils/getCookie';

const history = qhistory(
    createBrowserHistory({}),
    stringify,
    parse
);

const App = (props) => {

    if (!getCookie('token')) {
        sessionStorage.removeItem('lists');
        localStorage.removeItem('categories');

        return (
            <Router history={history}>
                <Header type='outer' history={history} />

                <Switch>
                    <Route exact path='/about' component={About} />
                    <Route exact path='/' component={Home} />
                    <Route
                        render={(props) => <NotFound type={'outer'} {...props} />}
                    />
                </Switch>

                <Footer type='outer' />
            </Router>
        );
    } else {
        return (
            <Router history={history}>
                <Header type='inner' history={history} />

                <Switch>
                    <Route 
                        exact path='/about' 
                        render={(props) => <About type={'inner'} {...props} />}
                    />
                    <Route exact path='/help' component={Help} />
                    <Route exact path='/settings' component={Settings} />
                    <Route exact path='/analytics' component={Analytics} />
                    <Route exact path='/archive' component={Archive} />
                    <Route exact path='/list/:id' component={List} />
                    <Route exact path='/' component={Lists} />
                    <Route
                        render={(props) => <NotFound type={'inner'} {...props} />}
                    />
                </Switch>

                <Footer type='inner' />
            </Router>         
        );
    }

};

export default connect(state => state)(App);
