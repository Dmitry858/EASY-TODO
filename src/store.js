import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import tasksStoreConfig from './reducers/tasks';

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return (handlers[action.type] && handlers[action.type](state, action)) || state;
    };
};

const tasksReducers = createReducer(tasksStoreConfig.initialState, tasksStoreConfig.actions);

const rootReducer = combineReducers({
    tasks: tasksReducers,
});

export default createStore(rootReducer, {}, applyMiddleware(thunk));