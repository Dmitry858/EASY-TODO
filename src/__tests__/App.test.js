import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import { Provider } from 'react-redux';
import store from '../store.js';
import { render } from '@testing-library/react';

describe('App testing', () => {
    it('App render without crashing', () => {
        const div = document.createElement('div');
      
        ReactDOM.render(
            <Provider store={store}>
                <App />
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('App matches the snapshot', () => {
        const { container } = render(
            <Provider store={store}>
                <App />
            </Provider>            
        );
        expect(container).toMatchSnapshot();
    });
    
});
