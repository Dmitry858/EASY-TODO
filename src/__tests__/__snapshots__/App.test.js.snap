// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing App matches the snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="container"
    >
      <div
        class="nav-wrapper"
      >
        <div
          class="logo-wrap"
        >
          <a
            aria-current="page"
            class="brand-logo active"
            href="/"
          >
            <img
              alt="Logo"
              src="logo-white.png"
            />
          </a>
        </div>
        <div
          class="main-menu-wrap"
        >
          <ul
            class="main-menu hide-on-med-and-down"
          >
            <li>
              <a
                class="sidenav-close"
                href="/about"
              >
                О сервисе
              </a>
            </li>
            <li>
              <a
                class="modal-trigger sidenav-close"
                href="#modal-signup"
              >
                Регистрация
              </a>
            </li>
            <li>
              <a
                class="modal-trigger sidenav-close"
                href="#modal-signin"
              >
                Вход
              </a>
            </li>
          </ul>
        </div>
        <a
          class="hide-on-large-only sidenav-trigger"
          data-target="SideNav-10"
          href="#"
        >
          <i
            aria-hidden="true"
            class="fa fa-bars"
          />
        </a>
        <ul
          class="sidenav"
          id="SideNav-10"
        >
          <li>
            <a
              class="sidenav-close"
              href="/about"
            >
              О сервисе
            </a>
          </li>
          <li>
            <a
              class="modal-trigger sidenav-close"
              href="#modal-signup"
            >
              Регистрация
            </a>
          </li>
          <li>
            <a
              class="modal-trigger sidenav-close"
              href="#modal-signin"
            >
              Вход
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div
    class="content"
  >
    <div
      class="container"
    >
      <div
        class="get-started row valign-wrapper"
      >
        <div
          class="col m6 center-align"
        >
          <h1
            class="center-align"
          >
            Приведите дела 
            <br />
            в порядок с 
            <br />
            <span>
              EASY TODO
            </span>
          </h1>
          <a
            class="btn btn-large modal-trigger waves-effect"
            href="#modal-signup"
          >
            Начать
          </a>
        </div>
        <div
          class="col m6 center-align"
        >
          <img
            alt="Приведите дела в порядок"
            src="image1.png"
          />
        </div>
      </div>
      <div
        class="screenshot"
      >
        <img
          alt="Приложение EASY TODO"
          src="screen.jpg"
        />
      </div>
    </div>
    <div
      class="read-more"
    >
      <div
        class="container center-align"
      >
        <p>
          Позвольте себе не думать постоянно о ежедневных заботах, переместите свои дела из головы в планировщик задач и обретите спокойствие.
        </p>
        <a
          href="/about"
        >
          <i
            aria-hidden="true"
            class="fa fa-arrow-right"
          />
          Узнать подробнее о приложении
        </a>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="relax row valign-wrapper"
      >
        <div
          class="col m6 center-align"
        >
          <img
            alt="Приведите дела в порядок"
            src="image2.png"
          />
        </div>
        <div
          class="col m6 center-align"
        >
          <h3
            class="center-align"
          >
            EASY TODO поможет организовать задачи наилучшим образом, чтобы у вас освободилось больше времени для отдыха
          </h3>
          <a
            class="btn btn-large modal-trigger waves-effect"
            href="#modal-signup"
          >
            Начать
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <p
        class="center-align"
      >
        Copyright © 2020 EASY TODO. Все права защищены.
      </p>
    </div>
  </footer>
</div>
`;
